app/scripts/directives/logViewer.js

```diff
@@ -446,17 +446,38 @@ angular.module('openshiftConsole')
                   access_token: AuthService.UserStore().getToken()
                 });
 
-                $scope.$watchGroup(['context.project.metadata.name', 'options.container', 'name'], function() {
-                  angular.extend($scope, {
-                    // The archive URL violates angular's built in same origin policy.
-                    // Need to explicitly tell it to trust this location or it will throw errors.
-                    kibanaArchiveUrl: $sce.trustAsResourceUrl(logLinks.archiveUri({
-                                        namespace: $scope.context.project.metadata.name,
-                                        namespaceUid: $scope.context.project.metadata.uid,
-                                        podname: name,
-                                        containername: $scope.options.container,
-                                        backlink: URI.encode($window.location.href)
-                                      }, $filter('annotation')($scope.context.project,'loggingDataPrefix')))
+                // var ssarVersion = APIService.getPreferredVersion('selfsubjectaccessreviews');
+                var ssar = {
+                  apiVersion: 'authorization.k8s.io/v1',
+                  // apiService: APIService.toAPIVersion(ssarVersion)
+                  kind: 'SelfSubjectAccessReview',
+                  spec: {
+                    resourceAttributes: {
+                      resource: 'pods/log',
+                      namespace: 'openshift-logging',
+                      verb: 'view'
+                    }
+                  }
+                };
+                var promise = DataService.create('selfsubjectaccessreviews', null, ssar, {namespace: 'openshift-logging'});
+                // var promise = DataService.create(ssarVersion, null, ssar, {namespace: 'openshift-logging'});
+                promise.then(function(data) {
+                  return data.status.allowed;
+                }, function() {
+                  return false;
+                }).then(function(canViewOperationsLogs) {
```

Comment:
@spadgett @jcantrill I'm struggling with the async nature of `promise`s. As this function is not called externally (click on a button etc), at this point log view is blocked. Can you help me how to make sure this promise gets evaluated?

Reply:
I had an issue with my VM, all log files disappeared from the file system. That's why I was seeing weird behavior in the console. The code should be fine.